src/database.js

```diff
@@ -42,17 +43,57 @@ exports.update = function (bill) {
 };
 
 exports.save = async function (watchlist) {
+	const writeFile = () => fs.writeFile(
+		`${databaseDirectory}${databaseFile}`,
+		JSON.stringify(database, null, '	')
+	).then(() => dirty = false);
+
+	const mkdirsAndWriteFile = async () => {
+		const dirParts = databaseDirectory.split('/').filter((d, j) => d.length > 0 || j === 0); // `j === 0` in case the start is at root
+		const joinDirParts = (i) => dirParts.filter((p, j) => j < i).join('/');
+		const mkdir = (i) => fs.mkdir(joinDirParts(i));
```

Comment:
The error handling is good but i'm not sure how this part is derived from node best practices.

Reply:
`joinDirParts` function was used to essentially make a listing of each node in the directory tree. I overlooked that `fs.mkdir` with the `recursive` option would make each of those nodes for you. Good catch- a lot of effort went into reinventing the wheel here, and probably for the worst.